provider/azure/azure_discover.go

```diff
@@ -115,7 +115,10 @@ func fetchAddrsWithTags(tagName string, tagValue string, vmnet network.Interface
 			continue
 		}
 		tv := (*v.Tags)[tagName] // *string
-		if tv == nil || *tv != tagValue {
+		if tv == nil {
+			continue
```

Comment:
Maybe add:

`l.Printf("[DEBUG] discover-azure: Interface %s did not have tag: %s", id, tagValue)`